airbyte-webapp/src/views/Connector/ServiceForm/serviceFormContext.tsx

```diff
@@ -83,18 +85,36 @@ export const ServiceFormContextProvider: React.FC<ServiceFormContextProviderProp
 
   const authFieldsToHide = useMemo(
     () =>
-      Object.values(serverProvidedOauthPaths(selectedConnector)).map((f) =>
-        makeConnectionConfigurationPath(f.path_in_connector_config)
-      ),
-    [selectedConnector]
+      isAuthFlowSelected
+        ? Object.values(serverProvidedOauthPaths(selectedConnector)).map((f) =>
+            makeConnectionConfigurationPath(f.path_in_connector_config)
+          )
+        : [],
+    [selectedConnector, isAuthFlowSelected]
   );
 
+  const authErrors = useMemo(() => {
+    //key of field path, value of error code
+    const authErrors: Record<string, string> = {};
+    authFieldsToHide.map((fieldString) => {
+      const meta = getFieldMeta(fieldString);
+      const { error } = meta;
+
+      if (submitCount > 0 && error) {
+        authErrors[fieldString] = error;
+      }
+
+      return authErrors;
+    });
+    return authErrors;
```

Comment:
Using `map` without using the result of it, is not really useful, since we cause JS to create a new array with all the return values in it, that we'll never use. We only want to iterate here, so this should rather be a `forEach` on the array. That said, this is the ideal scenario, where `reduce` is usually used in:


```ts
    return authFieldsToHide.reduce<Record<string, string>>((authErrors, fieldName) => {
      const { error } = getFieldMeta(fieldName);
      if (submitCount > 0 && error) {
        authErrors[fieldName] = error;
      }
      return authErrors;
    }, {});
```